### PR TITLE
🌱 Refine ironic provisioner log data format

### PR DESCRIPTION
**Pull request by sailorvii**
Refine the log data format from "1666158340.7038972" to "2022-10-19T05:45:40.674Z"

The original format is badly readable.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by honza** [comment]
I like this, happy to approve once conflicts are resolved.

**Comment by kashifest** [comment]
ow, this needs rebase again :( , sorry, can you please rebase ?

**Comment by kashifest** [comment]
Please remember to squash your commits also .

**Comment by sailorvii** [comment]
> Please remember to squash your commits also .

Hi, I've refined the conflict and squash the commits. Please help to check.
Thanks

**Comment by kashifest** [comment]
This is what I see in the logs

```
2022-12-20T10:12:18.397Z	INFO	baremetalhost-validation	validate create	{"name": "node-1"}
2022-12-20T10:12:18.398Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/validate-metal3-io-v1alpha1-baremetalhost", "code": 200, "reason": "", "UID": "ba340ed5-4a5b-422c-8a6d-67451b1c9a5d", "allowed": true}
2022-12-20T10:12:18.416Z	INFO	controllers.HostFirmwareSettings	start	{"hostfirmwaresettings": "metal3/node-0"}
2022-12-20T10:12:18.416Z	INFO	controllers.HostFirmwareSettings	could not get baremetalhost, not running reconciler	{"hostfirmwaresettings": "metal3/node-0"}
2022-12-20T10:12:18.422Z	INFO	controllers.BareMetalHost	done	{"baremetalhost": "metal3/node-1", "provisioningState": "deleting", "requeue": false, "after": "0s"}
2022-12-20T10:12:18.423Z	INFO	controllers.BareMetalHost	start	{"baremetalhost": "metal3/node-1"}
2022-12-20T10:12:18.423Z	INFO	controllers.BMCEventSubscription	start	{"bmceventsubscription": "metal3/node-1"}
2022-12-20T10:12:18.423Z	INFO	controllers.BMCEventSubscription	done	{"bmceventsubscription": "metal3/node-1"}
2022-12-20T10:12:18.430Z	INFO	controllers.HostFirmwareSettings	start	{"hostfirmwaresettings": "metal3/node-1"}
2022-12-20T10:12:18.430Z	INFO	controllers.HostFirmwareSettings	could not get baremetalhost, not running reconciler	{"hostfirmwaresettings": "metal3/node-1"}
```
 And I am happy. You will need to rebase it again. Sorry for back and forth so many times. Please unhold it once you have done it. I am approving it.
/approve
/hold

**Comment by sailorvii** [comment]
> This is what I see in the logs
> 
> ```
> 2022-12-20T10:12:18.397Z	INFO	baremetalhost-validation	validate create	{"name": "node-1"}
> 2022-12-20T10:12:18.398Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/validate-metal3-io-v1alpha1-baremetalhost", "code": 200, "reason": "", "UID": "ba340ed5-4a5b-422c-8a6d-67451b1c9a5d", "allowed": true}
> 2022-12-20T10:12:18.416Z	INFO	controllers.HostFirmwareSettings	start	{"hostfirmwaresettings": "metal3/node-0"}
> 2022-12-20T10:12:18.416Z	INFO	controllers.HostFirmwareSettings	could not get baremetalhost, not running reconciler	{"hostfirmwaresettings": "metal3/node-0"}
> 2022-12-20T10:12:18.422Z	INFO	controllers.BareMetalHost	done	{"baremetalhost": "metal3/node-1", "provisioningState": "deleting", "requeue": false, "after": "0s"}
> 2022-12-20T10:12:18.423Z	INFO	controllers.BareMetalHost	start	{"baremetalhost": "metal3/node-1"}
> 2022-12-20T10:12:18.423Z	INFO	controllers.BMCEventSubscription	start	{"bmceventsubscription": "metal3/node-1"}
> 2022-12-20T10:12:18.423Z	INFO	controllers.BMCEventSubscription	done	{"bmceventsubscription": "metal3/node-1"}
> 2022-12-20T10:12:18.430Z	INFO	controllers.HostFirmwareSettings	start	{"hostfirmwaresettings": "metal3/node-1"}
> 2022-12-20T10:12:18.430Z	INFO	controllers.HostFirmwareSettings	could not get baremetalhost, not running reconciler	{"hostfirmwaresettings": "metal3/node-1"}
> ```
> 
> And I am happy. You will need to rebase it again. Sorry for back and forth so many times. Please unhold it once you have done it. I am approving it. /approve /hold

Thanks for your patience to review this change.
I've rebased and please help to let it go.

Thanks,
WeiWei

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

File: main.go [hunk 1]
```diff
@@ -115,9 +116,15 @@ func main() {
 		"The address the health endpoint binds to.")
 	flag.IntVar(&webhookPort, "webhook-port", 9443,
 		"Webhook Server port (set to 0 to disable)")
+	opts := zap.Options{
+		Development: true,
```
Comment (lentzi90) [inline main.go:120]: Should we set this based on `devLogging` instead of hard coding it?
```suggestion
		Development: devLogging,
```

**Review by lentzi90 — COMMENTED** [review]
Nice improvement!
One question below
/hold

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Comment by sailorvii** [comment]
> /test-ubuntu-integration-main /test-centos-integration-main

Please help to remove the "/hold" flag if you have no other suggestions.

Thanks

**Comment by lentzi90** [comment]
@zaneb good point.
I think we should implement structured logging, as [recommended by CAPI](https://cluster-api.sigs.k8s.io/developer/logging.html) and also strive to follow the rest of the guidelines there. As I understand it this is also in alignment with Kubernetes and should overall make for less surprises and logs that are easier to parse/understand.

**Comment by zaneb** [comment]
I think #1249 will resolve this to everyone's satisfaction - it will do the right thing based on whether you passed the `-dev` flag.

**Pull request merged**